tests/st/utils/utils.py

```diff
@@ -73,3 +76,96 @@ def retry_until_success(function, retries=10, ex_class=Exception):
         else:
             # Successfully ran the function
             return result
+
+def check_number_endpoints(host, expected):
+    """
+    Check that a host has the expected number of endpoints in Calico
+    Parses the "calicoctl endpoint show" command for number of endpoints.
+    Raises AssertionError if the number of endpoints does not match the
+    expected value.
+
+    :param host: DockerHost object
+    :param expected: int, number of expected endpoints
+    :return:
+    """
+    hostname = get_hostname(host)
+    output = host.calicoctl("endpoint show")
+    lines = output.split("\n")
+    actual = 0
+
+    for line in lines:
+        columns = re.split("\s*\|\s*", line.strip())
+        if len(columns) > 1:
+            if columns[1] == hostname:
+                actual = columns[4]
+                break
+
+    if int(actual) != int(expected):
+        msg = "Incorrect number of endpoints: \n" \
+              "Expected: %s; Actual: %s" % (expected, actual)
+        raise AssertionError(msg)
+
+def check_profile(host, profile_name):
+    """
+    Check that profile is registered in Calico
+    Parse "calicoctl profile show" for the given profilename
+
+    :param host: DockerHost object
+    :param profile_name: String of the name of the profile
+    :return: Boolean: True if found, False if not found
+    """
+    output = host.calicoctl("profile show")
+    lines = output.split("\n")
+    found = False
+
+    for line in lines:
+        columns = re.split("\s*\|\s*", line.strip())
+        if len(columns) > 1:
+            if profile_name == columns[1]:
+                found = True
+
+    return found
+
+def get_profile_name(host, network):
+    """
+    Get the profile name from Docker
+    A profile is created in Docker for each Network object.
+    The profile name is a randomly generated string.
+
+    :param host: DockerHost object
+    :param network: Network object
+    :return: String: profile name
+    """
+    info_raw = host.execute("docker network inspect %s" % network.name)
+    info = json.loads(info_raw)
+    return info["id"]
+
+def get_hostname(host):
+    """
+    Get the hostname from Docker
+    The hostname is a randomly generated string.
+    Note, this function only works with a host with dind enabled.
+    Raises an exception if dind is not enabled.
+
+    :param host: DockerHost object
+    :return: hostname of DockerHost
+    """
+    hostname = docker.inspect("--format",
+                              "{{ .Config.Hostname }}",
+                              "%s" % host.name)
+    return hostname.strip()
+
+def check_network(host, network):
+    """
+    Checks that the given network is in Docker
+    Raises an exception if the network is not found
+
+    :param host: DockerHost object
+    :param network: Network object
+    :return: None
+    """
+    try:
+        host.execute("docker network inspect %s" % network.name)
+    except Exception as e:
+        raise e
```

Comment:
Probably better this write as as an assertion than making it throw a random exception...